core/templates/components/button-directives/primary-button.component.ts

```diff
@@ -13,49 +13,94 @@
 // limitations under the License.
 
 /**
- * @fileoverview Component for the primary buttons displayed on static pages.
+ * @fileoverview Component for primary buttons displayed on static pages.
  */
 
-import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
 import { downgradeComponent } from '@angular/upgrade/static';
-import { WindowRef } from
-  'services/contextual/window-ref.service';
 import './primary-button.component.css';
+import { WindowRef } from 'services/contextual/window-ref.service';
 
 
 @Component({
   selector: 'oppia-primary-button',
   templateUrl: './primary-button.component.html',
   styleUrls: ['./primary-button.component.css']
 })
-export class PrimaryButtonComponent {
+export class PrimaryButtonComponent implements OnInit {
+  /**
+   * The main text to be displayed on the button.
+   */
   @Input() buttonText: string = '';
+
+  /**
+   * An optional array of custom CSS classes to be applied to the component.
+   */
   @Input() customClasses?: string[];
+
+  /**
+   * The URL the button should navigate to when clicked, optional.
+   * If no buttonHref is provided then the component acts like an HTML button
+   * element instead of an anchor link element.
+   */
+  @Input() buttonHref: string = '#';
+
+  /**
+   * A boolean value indicating whether the button should be disabled, optional.
+   * Valid only when the component is an HTML button element and not an anchor.
+   */
   @Input() disabled?: boolean = false;
-  @Input() buttonHref: string | null = null;
+
+  /**
+   * The function to execute when the component is clicked, optional.
+   * The return type of the function must be void.
+   */
   @Output() onClickPrimaryButton: EventEmitter<void> = new EventEmitter<void>();
 
+  componentIsButton: boolean = false;
+  openInNewTab: boolean = false;
+
   constructor(
     private windowRef: WindowRef,
   ) {}
 
-  handleButtonClick(): void {
+  ngOnInit(): void {
+    this.componentIsButton = this.buttonHref === '#';
+  }
+
+  getButtonHref(): string {
+    return this.buttonHref;
+  }
+
+  getTarget(): string {
+    this.openInNewTab = this.isExternalLink(this.buttonHref);
+    return this.openInNewTab ? '_blank' : '_self';
+  }
+
+  private isExternalLink(link: string): boolean {
+    return link.startsWith('http://') || link.startsWith('https://');
+  }
+
+  handleButtonClick(event: MouseEvent): void {
+    event.preventDefault();
```

Comment:
Could you please add a comment explaining why we do this and linking to the relevant section in the gtag page you mentioned?

Reply:
Done